### PR TITLE
Update date validation in pension form

### PR DESCRIPTION
**Pull request by jbalboni**
This adds two new pieces of date validation to the pension form:

1. Date of birth must be before date entered service
2. Date of marriage must be before date of separation (for regular marriages and spouse marriages)

This also adds a migration to put users back on a page where the validation error exists.

screen shot 2017-10-09 at 3 03 42 pm
screen shot 2017-10-09 at 3 00 55 pm

### CODE REVIEW

**Comment by jbalboni** [comment]
@peggygannon Can you take a look at the validation messages?

File: test/pensions/migrations.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { expect } from 'chai';
+
+import migrations from '../../src/js/pensions/migrations';
+
+describe.only('Pension migrations', () => {
```
Comment (cvalarida) [inline test/pensions/migrations.unit.spec.js]: Probably don't want this anymore. 😉

File: test/pensions/migrations.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }],
+        veteranDateOfBirth: '2013-01-01'
+      },
+      metadata: {
+        returnUrl: 'asdf'
+      }
+    });
+
+    expect(metadata.returnUrl).to.equal('/military/history');
+    expect(formData).to.be.an('object');
```
Comment (cvalarida) [inline test/pensions/migrations.unit.spec.js]: Out of curiosity, why check for this?

**Review by cvalarida — COMMENTED** [review]
Looks good, but I'm wondering how it works if we save before the military service page, for example. We won't have the data that the migration looks for, so...I didn't look too hard at it, but the unit tests seem to show it's fine, so no worries.

Only reason it's not approved right here and now is because we can't let that `.only` into master. 🤖

File: test/pensions/migrations.unit.spec.js (hunk 1 above)
Comment (jbalboni) [reply test/pensions/migrations.unit.spec.js]: Bleh, forgot to push a commit.

File: test/pensions/migrations.unit.spec.js (hunk 2 above)
Comment (jbalboni) [reply test/pensions/migrations.unit.spec.js]: I want to make sure we don't accidentally forget to return `formData`.

File: src/js/pensions/validation.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    errors.addError('Date marriage ended must be after date of marriage');
+  }
+}
+
+export function validateServiceBirthDates(errors, service, formData) {
+  const fromDate = convertToDateField(formData.veteranDateOfBirth);
+  const toDate = convertToDateField(_.get('activeServiceDateRange.from', service));
+
+  if (!isValidDateRange(fromDate, toDate)) {
+    errors.activeServiceDateRange.from.addError('Date entering active service should be after birth date');
```
Comment (cehsu) [inline src/js/pensions/validation.js]: Is legal enlistment age, also being checked here? If not, would it make sense to?

**Review by cvalarida — APPROVED** [review]
Beautiful!

File: src/js/pensions/config/form.js [hunk 4]
```diff
@@ -60,6 +60,9 @@ import otherExpensesUI from '../definitions/otherExpenses';
 import currencyUI from '../../common/schemaform/definitions/currency';
 import GetFormHelp from '../../common/schemaform/GetPensionOrBurialFormHelp';
 
+import { validateServiceBirthDates, validateAfterMarriageDate } from '../validation';
```
Comment (cehsu) [inline src/js/pensions/config/form.js]: Would it make sense to also check child DOB?

**Review by cehsu — APPROVED** [review]
LGTM. Had a few minor questions about scope of changes, which if relevant could be converted into new tickets.

**Comment by jbalboni** [comment]
@cehsu Those could be useful. I think the issue with date validation is that we can almost always add more checks, but eventually it becomes confusing to both us and users when something goes wrong. If you want to create a ticket with those suggestions that's fine with me, I just want to make sure @stacyhlinn thinks they would be valuable checks before we add them.

**Comment by jbalboni** [comment]
@cvalarida To answer your question about saving before military history, the migration has a check for each of those arrays, and also the date range validation passes by default if one of the dates is empty. So we shouldn't hit any problems if users don't get to a particular page or opt not to fill in some information.

**Pull request merged**